src/Middleware/ErrorHandler.php

```
@@ -0,0 +1,216 @@
+<?php
+/**
+ * @link      http://github.com/zendframework/zend-stratigility for the canonical source repository
+ * @copyright Copyright (c) 2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Stratigility\Middleware;
+
+use ErrorException;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Throwable;
+use Zend\Escaper\Escaper;
+use Zend\Stratigility\Exception\MissingResponseException;
+use Zend\Stratigility\Utils;
+
+class ErrorHandler
+{
+    /**
+     * @var bool
+     */
+    protected $isDevelopmentMode;
+
+    /**
+     * @var callable[]
+     */
+    private $listeners = [];
+
+    /**
+     * @var ResponseInterface
+     */
+    private $responsePrototype;
+
+    /**
+     * @param ResponseInterface $responsePrototype Empty/prototype response to
+     *     update and return when returning an error response.
+     * @param bool $isDevelopmentMode
+     */
+    public function __construct(ResponseInterface $responsePrototype, $isDevelopmentMode = false)
```

I've actually just pushed changes based on the feedback from @vaclavvanik that shoves that stuff into the response generator. As such, the `ErrorHandler` now has only the following responsibilities:

- compose a response prototype, a response generator, and listeners
- create and enable a PHP error handler that casts errors to exceptions
- detect when no inner middleware returns a response, and raise an exception
- catch all exceptions, and pass them, the request, and the prototype response to the response generator
- notifiy all listeners of the error, the request, and the final response

As such, no "optional" types of behavior are present, and those would be handled by the response generator and/or listeners, which fall out of scope of the class itself.